From 94638bfbea5da77f3a291c63c15446badd5210b5 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Fri, 8 Nov 2013 05:46:26 +0530
Subject: Set Stream / parameter flag in route manger to enable offload

BZ: 150633

This patch changes the way policy and primary HAL (so Route manager)
communicate. It now use a setParameter with a StreamFlag key
when a compressed stream starts and stops.

Category: Feature Differentiation
Domain: AUDIO_MEDIA-Audio Policy-LPA
Origin: Internal
Upstream-Candidate: no, proprietary

Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
Change-Id: I4be631b8ca026703a794b2a984728090bf8efa80
---
 audio/AudioPolicyManagerBase.cpp |   17 ++++++++++++++++-
 1 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/audio/AudioPolicyManagerBase.cpp b/audio/AudioPolicyManagerBase.cpp
index 269c274..00f2bfc 100644
--- a/audio/AudioPolicyManagerBase.cpp
+++ b/audio/AudioPolicyManagerBase.cpp
@@ -787,7 +787,15 @@ status_t AudioPolicyManagerBase::startOutput(audio_io_handle_t output,
 
     AudioOutputDescriptor *outputDesc = mOutputs.valueAt(index);
 
-    // increment usage count for this stream on the requested output:
+    if (outputDesc->mFlags & AUDIO_OUTPUT_FLAG_COMPRESS_OFFLOAD) {
+    // Informs primary HAL that a compressed output will be started
+        AudioParameter param;
+        param.addInt(String8(AudioParameter::keyStreamFlags), AUDIO_OUTPUT_FLAG_COMPRESS_OFFLOAD);
+        mpClientInterface->setParameters(0, param.toString(), 0);
+    // Stores the current audio sessionId for use in gapless offlaoded playback.
+    // mMusicOffloadSessionId = session;
+    }
+    //increment usage count for this stream on the requested output:
     // NOTE that the usage count is the same for duplicated output and hardware output which is
     // necessary for a correct control of hardware output routing by startOutput() and stopOutput()
     outputDesc->changeRefCount(stream, 1);
@@ -865,6 +873,13 @@ status_t AudioPolicyManagerBase::stopOutput(audio_io_handle_t output,
     }
 
     if (outputDesc->mRefCount[stream] > 0) {
+        if (outputDesc->mFlags & AUDIO_OUTPUT_FLAG_COMPRESS_OFFLOAD) {
+            // Informs primary HAL that a compressed output stops
+            AudioParameter param;
+            param.addInt(String8(AudioParameter::keyStreamFlags), AUDIO_OUTPUT_FLAG_NONE);
+            mpClientInterface->setParameters(0, param.toString(), 0);
+        }
+
         // decrement usage count of this stream on the output
         outputDesc->changeRefCount(stream, -1);
         // store time at which the stream was stopped - see isStreamActive()
-- 
1.7.4.1

